File: exercise1.js
let tries = 3
let balance = 2000
let balance_2 = 2000 
let user1 = "admin"
let pass1 = "1234"
let user2 = "admin2"
let pass2 = "5678"

let bool1 = false
let bool2 = false


function login(){
   
    if((user1 == document.getElementById("user").value && pass1 == document.getElementById("pass").value)){
        alert("Correct Data. Access Granted.")
        document.getElementById('bank').hidden = false   
        bool1 = true  
        bool2 = false
        document.getElementById('p-1').innerHTML = "Balance: " + balance
        document.getElementById('p-1').hidden = true; 
        document.getElementById('deposit-section').hidden = true;
        document.getElementById('transfer-section').hidden = true;
        document.getElementById('withdraw-section').hidden = true;
        document.getElementById('console').innerHTML = ""
    }
    else if(user2 == document.getElementById("user").value && pass2 == document.getElementById("pass").value){
        alert("Correct Data. Access Granted.")
        document.getElementById('bank').hidden = false
        bool2 = true   
        bool1 = false 
        document.getElementById('p-1').innerHTML = "Balance: " + balance_2
        document.getElementById('p-1').hidden = true; 
        document.getElementById('deposit-section').hidden = true;
        document.getElementById('transfer-section').hidden = true;
        document.getElementById('withdraw-section').hidden = true;
        document.getElementById('console').innerHTML = ""
    }
    else{
        tries--
        alert("Wrong Data. " + tries + " tries left")
        if(tries == 0){
            alert("Access Denied.")
            location.href ='https://google.com';
        }
    }
}

function view(){
    document.getElementById('console').innerHTML = "Account view:"
    if(bool1){
        document.getElementById('p-1').innerHTML = "Balance: " + balance
    }
    else if(bool2){
        document.getElementById('p-1').innerHTML = "Balance: " + balance_2
    }
    document.getElementById('p-1').hidden = false; 
    document.getElementById('deposit-section').hidden = true; 
    document.getElementById('transfer-section').hidden = true;
    document.getElementById('withdraw-section').hidden = true; 
    document.getElementById('console-2').innerHTML = ""  
    
}

function deposit(){
    document.getElementById('console').innerHTML = "Deposit view:"
    document.getElementById('p-1').innerHTML = "Enter the amount to deposit in your account: " 
    document.getElementById('deposit-section').hidden = false; 
    document.getElementById('transfer-section').hidden = true;
    document.getElementById('withdraw-section').hidden = true; 
    document.getElementById('console-2').innerHTML = ""  
}

function rdeposit(){
    if(bool1 ){
        balance += parseInt(document.getElementById("deposit-input").value)          
    }
    else if(bool2 ){
        balance_2 += parseInt(document.getElementById("deposit-input").value)     
    } 
    document.getElementById('console-2').innerHTML = "Successful deposit."   
}

function transfer(){
    document.getElementById('console').innerHTML = "Transfer view:"
    document.getElementById('p-1').innerHTML = "Enter the amount to transfer from the other account: " 
    document.getElementById('deposit-section').hidden = true;
    document.getElementById('transfer-section').hidden = false;
    document.getElementById('withdraw-section').hidden = true;
    document.getElementById('console-2').innerHTML = ""        
}

function rtransfer(){
    if(bool1){
        balance_2 += parseInt(document.getElementById("transfer-input").value)
        balance -= parseInt(document.getElementById("transfer-input").value)            
    }
    else if(bool2){
        balance += parseInt(document.getElementById("transfer-input").value)   
        balance_2 -= parseInt(document.getElementById("transfer-input").value)      
    }
    document.getElementById('console-2').innerHTML = "Successful transfer."   
       
}

function withdraw(){
    document.getElementById('console').innerHTML = "Whitdraw view:"
    document.getElementById('p-1').innerHTML = "Enter the amount to witdraw from your account: " 
    document.getElementById('deposit-section').hidden = true;
    document.getElementById('transfer-section').hidden = true;
    document.getElementById('withdraw-section').hidden = false;      
    document.getElementById('console-2').innerHTML = ""  
}

function rwithdraw(){
    if(bool1 ){
        balance -= parseInt(document.getElementById("withdraw-input").value)          
    }
    else if(bool2 ){
        balance_2 -= parseInt(document.getElementById("withdraw-input").value)     
    }    
    document.getElementById('console-2').innerHTML = "Successful whitdraw."   
}

function exit(){
    alert('Thanks for using our services, have a nice day')
    document.getElementById('bank').hidden = true;  

}

File: Exercise1.py
#Exercise 1#
#!Create an online banking system!#
#|Users must be able to log in with a username and password
#|If the user enters the wrong credentials three times, the system must lock them out
#|The initial balance in the bank account is $2000
#|The system must allow users to deposit, withdraw, view, and transfer money
#|The system must display a menu for users to perform transactions

users = {"admin": "12345", "admin2":"67890"}
balance = 2000

def login_screen():
    username = input("Please enter your username: ")
    password = input("Please enter your password: ")
    return username, password

print("Hello, welcome to our bank")
username_input, password_input = login_screen()

def deposit(balance):
    dep = input("Insert the amount you want to deposit: $")
    if(dep.isnumeric()):
        return main_screen(balance + int(dep))
    else:
        print("The amount is not valid")
        return main_screen(balance)

def withdraw(balance):
    wit = input("Insert the amount you want to withdraw: $")
    if(wit.isnumeric()):
        if(balance - int(wit) < 0):
            print("You can't withdraw that amount")
            return main_screen(balance)
        else:
            return main_screen(balance - int(wit))
    else:
        print("The amount is not valid")
        return main_screen(balance)

def transfer(balance):
    transfer_user = input("Insert the username you want to transfer to: ")
    if(transfer_user in users):
        print("The user was found")
        tra = input(f"Insert the amount you want to transfer to the user with the card \"{transfer_user}\": $")
        if(tra.isnumeric()):
            if(balance - int(tra) <= 0):
                print("You can't transfer that amount")
                return main_screen(balance)
            else:
                return main_screen(balance - int(tra))
        else:
            print("The amount is not valid")
            return main_screen(balance)
    else:
        print("The user wasn't found")
        return main_screen(balance)

def main_screen(actual_balance):
    print("You can do the next actions: \n1.Deposit \n2.Withdraw \n3.View \n4.Transfer \n5.Exit")
    action = input("What would you like to do?: ")
    match action:
        case "1":
            deposit(actual_balance)
        case "2":
            withdraw(actual_balance)
        case "3":
            print(f"Current balance: ${actual_balance}")
            main_screen(actual_balance)
        case "4":
            transfer(actual_balance)
        case "5":
            print("Thanks for using the online banking system")
        case _:
            print("This action doesn't exist. Please try with another action")
            main_screen()

failed_attempts = 3
while(failed_attempts > 1):
    if(username_input in users):
        if(password_input == users[username_input]):
            print("Welcome")
            main_screen(balance)
            break
        else:
            failed_attempts -= 1
            print("Incorrect password. Try Again")
            username_input, password_input = login_screen()
    else:
        failed_attempts -= 1
        print("Username doesn't exist. Try Again")
        username_input, password_input = login_screen()
else:
    print("Too many failed attempts, f**k off")